### PR TITLE
Add plan for rebar3

### PR DESCRIPTION
**Pull request by elliott-davis**
Signed-off-by: Elliott Davis <edavis@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
By analyzing the blame information on this pull request, we identified @adamhjk to be a potential reviewer

File: plans/rebar3/plan.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+pkg_deps=(core/erlang)
+pkg_build_deps=(core/coreutils)
+pkg_bin_dirs=(bin)
+pkg_lib_dirs=(lib)
+
+do_prepare() {
+  # The `/usr/bin/env` path is hardcoded, so we'll add a symlink if needed.
+  if [[ ! -r /usr/bin/env ]]; then
+    ln -sv $(pkg_path_for coreutils)/bin/env /usr/bin/env
+    _clean_env=true
```
Comment (metadave) [inline plans/rebar3/plan.sh]: this appears to be unused

**Comment by metadave** [comment]
```

**Comment by metadave** [comment]
looks like this needs a runtime dep for `sed`:
```
root@5416bc1c1e72:/src/plans# hab pkg exec core/rebar3 rebar3
/hab/pkgs/core/erlang/18.3/20160612151228/bin/erl: 1: /hab/pkgs/core/erlang/18.3/20160612151228/bin/erl: sed: not found
```

File: plans/rebar3/plan.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  fi
+}
+
```
Comment (fnichol) [inline plans/rebar3/plan.sh]: Possible to remove the one extra newline?

**Comment by fnichol** [comment]
When I ran the rebar3 plan and examined `bin/rebar3` I found that it was largely binary, but thinly wrapped with a shebang:

```sh
[11][default:/hab/pkgs/core/rebar3/3.1.1/20160624160040/bin:0]# head -n 2 rebar3
#!/usr/bin/env escript
%%
```

Given that strictly inside Habitat, we can't guarantee that `/usr/bin/env` exists, it's probably worth changing this shebang. There are 2 ways to go, in my mind:

1. Taking a dependency on `core/busybox-static`, `core/coreutils-static`, or `core/coreutils` in order to to use the [`fix_interpreter()`](https://www.habitat.sh/docs/reference/plan-syntax/#utility-functions) helper function
2. Update the shebang to directly run `escript` as there isn't really a downside to doing this directly with an absolute path. The resulting value would be something like: `#!${pkg_path_for erlang}/bin/escript`. This strategy is used in the [node Plan](https://github.com/habitat-sh/habitat/blob/master/plans/node/plan.sh#L35-L40).

**Comment by elliott-davis** [comment]
good catch on that binary @fnichol - I got it patched up using fix_interpreter. I also got the newline removed.

File: plans/rebar3/plan.sh (hunk 1 above)
Comment (elliott-davis) [inline plans/rebar3/plan.sh]: updated so that it's used now

File: plans/rebar3/plan.sh (hunk 2 above)
Comment (elliott-davis) [inline plans/rebar3/plan.sh]: removed

**Comment by metadave** [comment]
@thesentinels delegate=elliott-davis

I'm delegating merge access to you, please read [these instructions](https://github.com/habitat-sh/habitat#delegating-pull-request-merge-access) for more info!

**Comment by thesentinels** [comment]
:v: @elliott-davis can now approve this pull request

**Comment by elliott-davis** [comment]
@thesentinels r+

**Comment by thesentinels** [comment]
:pushpin: Commit 191022c has been approved by `elliott-davis`

**Comment by thesentinels** [comment]
:hourglass: Testing commit 191022c with merge 59dcaa6...

**Comment by thesentinels** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/habitat-sh/habitat/builds/141367784)

**Comment by smith** [comment]
Built and uploaded to depot https://app.habitat.sh/#/pkgs/core/rebar3/3.1.1/20160630160805

**Pull request merged**